rumcake/src/hw/mod.rs

```diff
@@ -123,6 +127,22 @@ pub async fn output_switcher() {
     }
 }
 
+#[cfg(feature = "bootloader-double-tap-reset")]
+pub static BOOTLOADER_MAGIC: u32 = 0xDEADBEEF;
+
+#[cfg(feature = "bootloader-double-tap-reset")]
+#[link_section = ".uninit"]
+pub static mut FLAG: GroundedCell<u32> = GroundedCell::uninit();
+
+// inspired by https://github.com/qmk/qmk_firmware/blob/master/platforms/chibios/bootloaders/rp2040.c
+#[cfg(feature = "bootloader-double-tap-reset")]
+#[rumcake_macros::task]
+pub async unsafe fn clear_bootloader_magic_task(timeout: u64) {
```

Comment:
I think it's better if this just a normal async function instead of a task.
```suggestion
pub async unsafe fn clear_bootloader_magic_task(timeout: u64) {
```

Be sure to remove the export from the `tasks` module as well.

Reply:
Also `u64` might be a bit overkill here, I think `u16` would be sufficient.

Reply:
changed to an async function, rumcake::hw is a good location for that function? 

The `u64` is overkill. But that's because [Timer::after_millis()](https://docs.rs/embassy-time/latest/embassy_time/struct.Duration.html#method.from_millis) takes a `u64` so the value would need to be casted to a `u64` anyways in the end.

Reply:
Ah my bad, I missed that.

Reply:
>  is rumcake::hw still the right location for that function?

Also, yes it can be kept in there.